From d6abb8786b3c614bf0d3667c14ac178f250b3df0 Mon Sep 17 00:00:00 2001
From: Nigel Cunningham <nigel@tuxonice.net>
Date: Mon, 10 Mar 2014 21:01:13 +1100
Subject: [PATCH] Fix VM_BUG_ON in swap allocation.

The VM_BUG_ON was being triggered because our invocation of
inc_cluster_info_page for each page in a range of pages being allocated
was including the first page. The first page has already had
inc_cluster_info_page via scan_swap_map.

Signed-off-by: Nigel Cunningham <nigel@tuxonice.net>
---
 mm/swapfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/swapfile.c b/mm/swapfile.c
index cf86848..4044ab6 100644
--- a/mm/swapfile.c
+++ b/mm/swapfile.c
@@ -774,7 +774,7 @@ void get_swap_range_of_type(int type, swp_entry_t *start, swp_entry_t *end,
 				si->lowest_bit = si->max;
 				si->highest_bit = 0;
 			}
-			for (i = start_at; i <= stop_at; i++)
+			for (i = start_at + 1; i <= stop_at; i++)
 				inc_cluster_info_page(si, si->cluster_info, i);
 			si->cluster_next = stop_at + 1;
 			*start = swp_entry(type, start_at);
-- 
1.8.5.5

